### PR TITLE
enable assert() in release builds

### PR DESCRIPTION
**Pull request by jlevon**
Also fix release build failures, and build in release mode in travis.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -32,9 +32,6 @@ cmake_minimum_required (VERSION 2.6)
 project(muser)
 include(GNUInstallDirs)
 
-
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Werror -Wextra")
```
Comment (swapnili) [inline CMakeLists.txt:36]: Why to remove this? We do want to catch the compiler warnings all over the code.

File: lib/CMakeLists.txt [hunk 2]
```diff
@@ -30,6 +30,11 @@
 
 set(CMAKE_C_FLAGS "-Wall -Wextra -Werror -fPIC")
 set(CMAKE_C_FLAGS_DEBUG "-O0 -ggdb")
+#
+# NB: this intentionally does not define NDEBUG: we want our assert()s even in a
+# release build.
+#
+set(CMAKE_C_FLAGS_RELEASE "-O3")
```
Comment (swapnili) [inline lib/CMakeLists.txt:37]: I think NDEBUG should be undefined from top level CMakeLists.txt. So that this rule is applicable all over the repo.

**Review by swapnili — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (jlevon) [reply CMakeLists.txt:36]: Yeah, this was a mistake; I've reworked how we set the flags in both places.

File: lib/CMakeLists.txt (hunk 2 above)
Comment (jlevon) [reply lib/CMakeLists.txt:37]: Thanks, yes!

**Review by swapnili — APPROVED** [review]
The travis build failed. Looks like you will need to rebase on master.

**Comment by jlevon** [comment]
> The travis build failed. Looks like you will need to rebase on master.

Thanks, yes, I'll need to rebase the series.

**Review by swapnili — APPROVED** [review]

**Pull request merged**